developer_workflow.md

```diff
@@ -1,41 +1,87 @@
-* Verify everything works locally with real data. Play with the feature and ensure the user experience is good.
-* Add to the README as you develop, making note of any steps required for deployment and testing: migrations, running scripts, updates to the server, etc. Add this information to your card, in addition to the README. See our (readme document)[https://github.com/HaughtCodeworks/TeamGuide] for more on this.
-* Consider if you need to break your changes into multiple commits. Once you've grouped your changes into logical commits, add a good commit message. Make sure that you are only committing the changes that make sense for this feature.
-* Push your branch to origin, and avoid including sensitive files in source control.
+Each project may vary so this guide describes our standard workflow.
+
+# Assigning Work
+
+Each project will have a weekly planning meeting to go over what's new on the project, recently completed items, and what the team will work on in the coming week.  We cover the elements of this meeting in more detail in our [Project Planning Meeting guide](https://github.com/HaughtCodeworks/TeamGuide).  We'll focus here on the developer-centric elements of work once it has been assigned.
+
+# Card Review
+
+During the planning meeting, the team will discuss with the client each item (which we refer to as a card) to be worked on.  Before we can proceed with any development we need to review each card, covering the card review elements below.  Our goal is to keep as much discussion as possible in the planning meeting, trying to avoid the need to have additional meetings, email exchanges or chat in order for a developer to start and complete work on the card.  Not every card will need this much discussion but consider each of these items and determine if it's appropriate to bring into the discussion.  Again, we don't want to waste time but we also want to avoid misunderstandings or miscommunication about what we need to do on the card.
+
+## Card Review Elements
+
+Understand what the client is asking for.  The client should explain not only the business purpose of the change but any set expectations.
+
+What are the business requirements? Any existing specifications, technical requirements, mock-ups, or other supporting documents we should review?
+
+If the card represents a bug or defect, ask how it can be reproduced as well as the severity of the defect.
+
+Are there target dates either for initial review or going live with the change?  If so, find out more about the nature of those dates.  Are they fixed or flexible?  What's the consequence of missing the date?  Are they vital enough that the scope of the requested work should shift?
+
+Ask questions to clarify anything that you either don't understand or isn't clear in what has been covered so far.
+
+Explore edge case scenarios that may not have been covered to see how the client wants those resolved.
+
+Clarify where in the system these changes should take place.
```

Comment:
I'm thinking we may need to expand on this one with something like this:

If you are not 100% certain, clarify where in the application these changes should take place. For example, in more complex applications similar functionality may occur on different pages for different user types.  If there's any ambiguity make sure you clarify.